ansible_ai_connect/ai/api/views.py

```diff
@@ -337,40 +353,14 @@ def write_to_segment(
                     ansible_extension_version,
                 )
 
-        feedback_events = [
-            inline_suggestion_data,
-            suggestion_quality_data,
-            sentiment_feedback_data,
-            issue_feedback_data,
-        ]
-        if exception and all(not data for data in feedback_events):
-            # When an exception is thrown before inline_suggestion_data or ansible_content_data
-            # is set, we send request_data to Segment after having anonymized it.
-            ano_request_data = anonymizer.anonymize_struct(request_data)
-            if "inlineSuggestion" in request_data:
-                event_type = "inlineSuggestionFeedback"
-            elif "suggestionQualityFeedback" in request_data:
-                event_type = "suggestionQualityFeedback"
-            elif "sentimentFeedback" in request_data:
-                event_type = "sentimentFeedback"
-            elif "issueFeedback" in request_data:
-                event_type = "issueFeedback"
-            else:
-                event_type = "unknown"
-
-            event = {
-                "data": ano_request_data,
-                "exception": str(exception),
-            }
-            send_segment_event(event, event_type, user)
-
 
 class ContentMatches(GenericAPIView):
```

Comment:
Should this extend `OurAPIView` too?

You've [removed](https://github.com/ansible/ansible-ai-connect-service/pull/1147/files#diff-ecfb6919dfd8379aafba96af7457b253e4dce528897dfe6bfc207ca2b3b2ada9L550-L573) the code sending segment events; so presumably expect `OurAPIView` to be handling it?

You could also then be able to remove [the](https://github.com/ansible/ansible-ai-connect-service/pull/1147/files#diff-ecfb6919dfd8379aafba96af7457b253e4dce528897dfe6bfc207ca2b3b2ada9R396-R399) de-serialization?

Reply:
Yes, indeed.